pkg/controller/groups/variables/controller.go

```diff
@@ -0,0 +1,233 @@
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane-contrib/provider-gitlab/apis/groups/v1alpha1"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients/groups"
+)
+
+const (
+	errNotVariable       = "managed resource is not a Gitlab variable custom resource"
+	errGetFailed         = "cannot get Gitlab variable"
+	errCreateFailed      = "cannot create Gitlab variable"
+	errUpdateFailed      = "cannot update Gitlab variable"
+	errDeleteFailed      = "cannot delete Gitlab variable"
+	errGetSecretFailed   = "cannot get secret for Gitlab variable value"
+	errSecretKeyNotFound = "cannot find key in secret for Gitlab variable value"
+	errGroupIDMissing    = "GroupID is missing"
+)
+
+// SetupVariable adds a controller that reconciles Variables.
+func SetupVariable(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.VariableKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Variable{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.VariableGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newGitlabClientFn: groups.NewVariableClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube              client.Client
+	newGitlabClientFn func(cfg clients.Config) groups.VariableClient
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.Variable)
+	if !ok {
+		return nil, errors.New(errNotVariable)
+	}
+	cfg, err := clients.GetConfig(ctx, c.kube, cr)
+	if err != nil {
+		return nil, err
+	}
+	return &external{kube: c.kube, client: c.newGitlabClientFn(*cfg)}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client groups.VariableClient
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Variable)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotVariable)
+	}
+	if cr.Spec.ForProvider.GroupID == nil {
+		return managed.ExternalObservation{}, errors.New(errGroupIDMissing)
+	}
+
+	variable, res, err := e.client.GetVariable(
+		*cr.Spec.ForProvider.GroupID,
+		cr.Spec.ForProvider.Key,
+		gitlab.WithContext(ctx))
+
+	if err != nil {
+		if clients.IsResponseNotFound(res) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetFailed)
+	}
+
+	if cr.Spec.ForProvider.ValueSecretRef != nil {
+		if err = e.updateVariableFromSecret(ctx, cr.Spec.ForProvider.ValueSecretRef, &cr.Spec.ForProvider); err != nil {
+			cr.Status.SetConditions(xpv1.ReconcileError(err))
```

Comment:
This is not necessary. The ReconcileError status condition is set automatically when you return an error during `Oberve`, `Create`, `Update` and `Delete`.

```suggestion
```